server/sampling.go

```diff
@@ -0,0 +1,56 @@
+package server
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mark3labs/mcp-go/mcp"
+)
+
+// SamplingClient defines the interface for servers to request sampling from clients.
+type SamplingClient interface {
+	// RequestSampling sends a sampling request to the client and returns the result.
+	RequestSampling(ctx context.Context, request mcp.CreateMessageRequest) (*mcp.CreateMessageResult, error)
+}
+
+// EnableSampling enables sampling capabilities for the server.
+// This allows the server to send sampling requests to clients that support it.
+func (s *MCPServer) EnableSampling() {
+	s.capabilitiesMu.Lock()
+	defer s.capabilitiesMu.Unlock()
+
+	if s.capabilities.sampling == nil {
+		s.capabilities.sampling = &samplingCapabilities{}
+	}
+	s.capabilities.sampling.enabled = true
+}
+
+// RequestSampling sends a sampling request to the client.
+// The client must have declared sampling capability during initialization.
+func (s *MCPServer) RequestSampling(ctx context.Context, request mcp.CreateMessageRequest) (*mcp.CreateMessageResult, error) {
+	s.capabilitiesMu.RLock()
+	enabled := s.capabilities.sampling != nil && s.capabilities.sampling.enabled
+	s.capabilitiesMu.RUnlock()
+
+	if !enabled {
+		return nil, fmt.Errorf("sampling not enabled on server")
+	}
+
+	session := ClientSessionFromContext(ctx)
+	if session == nil {
+		return nil, fmt.Errorf("no active session")
+	}
+
+	// Check if the session supports sampling requests
+	if samplingSession, ok := session.(SessionWithSampling); ok {
```

Comment:
It would be better if we could find a way to ensure that session is an implementation of SessionWithSampling when `EnableSampling` is invoked.